presto-hive-common/src/main/java/com/facebook/presto/hive/HiveFileContext.java

```diff
@@ -17,29 +17,55 @@
 import com.facebook.presto.common.RuntimeUnit;
 
 import java.util.Optional;
+import java.util.OptionalLong;
 
 import static com.facebook.presto.hive.CacheQuota.NO_CACHE_CONSTRAINTS;
 import static java.util.Objects.requireNonNull;
 
 public class HiveFileContext
 {
-    public static final HiveFileContext DEFAULT_HIVE_FILE_CONTEXT = new HiveFileContext(true, NO_CACHE_CONSTRAINTS, Optional.empty(), Optional.empty(), 0, false);
+    public static final HiveFileContext DEFAULT_HIVE_FILE_CONTEXT = new HiveFileContext(
+            true,
+            NO_CACHE_CONSTRAINTS,
+            Optional.empty(),
+            OptionalLong.empty(),
+            OptionalLong.empty(),
+            OptionalLong.empty(),
+            0,
+            false);
 
     private final boolean cacheable;
     private final CacheQuota cacheQuota;
     private final Optional<ExtraHiveFileInfo<?>> extraFileInfo;
-    private final Optional<Long> fileSize;
+    private final OptionalLong fileSize;
+    // HiveFileContext optionally contains startOffset and length that readers are going to read
+    // For large files, these hints will help in fetching only the required blocks from the storage
+    // Note: even when they are present, readers can read past this range like footer and stripe's
+    // start may be in this range, but end may be beyond this range. So this is a hint and readers
+    // should handle the read beyond the range gracefully.
+    private final OptionalLong startOffset;
+    private final OptionalLong length;
     private final long modificationTime;
     private final boolean verboseRuntimeStatsEnabled;
 
     private final RuntimeStats stats = new RuntimeStats();
 
-    public HiveFileContext(boolean cacheable, CacheQuota cacheQuota, Optional<ExtraHiveFileInfo<?>> extraFileInfo, Optional<Long> fileSize, long modificationTime, boolean verboseRuntimeStatsEnabled)
+    public HiveFileContext(
+            boolean cacheable,
+            CacheQuota cacheQuota,
+            Optional<ExtraHiveFileInfo<?>> extraFileInfo,
+            OptionalLong fileSize,
+            OptionalLong startOffset,
```

Comment:
nit: just `offset` would be fine too

Reply:
some places use offset and other places use start, so I used startOffset :)